### PR TITLE
Add section describe spot instance reclamation

### PR DESCRIPTION
**Pull request by jradtilbrook**
Add a troubleshooting section on spot instance reclamation in the elastic stack

### CODE REVIEW

File: pages/agent/v3/elastic_ci_aws/troubleshooting.md [hunk 1]
```diff
@@ -35,6 +35,12 @@ Resource shortage can cause this issue. See the Auto Scaling group's Activity lo
 
 To fix this issue, change or add more instance types to the `InstanceTypes` template parameter. If 100% of your existing instances are Spot Instances, switch some of them to On-Demand Instances by setting `OnDemandPercentage` parameter to a value above zero.
 
+## Instances are abruptly terminated
+
+This can happen when using Spot Instances. AWS EC2 sends a notification to a spot instance 2 minutes prior to termination. The agent intercepts that notification and attempts to gracefully shutdown. If the instance does not shut down gracefully in that time, it is terminated.
+
+If 100% of your existing instances are Spot Instances, switch some of them to On-Demand Instances by setting `OnDemandPercentage` parameter to a value above zero.
```
[the remark this answers is not in the document]
Comment (mbelton-buildkite) [reply pages/agent/v3/elastic_ci_aws/troubleshooting.md:42]: Actually, is there something missing here? It reads as if you should always have some on-demand instances but I assume it's valid to not. Should it be "If all your your existing instances are Spot Instances and [an extra condition], switch..."

[the remark this answers is not in the document]
Comment (jradtilbrook) [reply pages/agent/v3/elastic_ci_aws/troubleshooting.md:42]: No I think your suggestion fits

File: pages/agent/v3/elastic_ci_aws/troubleshooting.md [hunk 2]
```diff
@@ -35,6 +35,12 @@ Resource shortage can cause this issue. See the Auto Scaling group's Activity lo
 
 To fix this issue, change or add more instance types to the `InstanceTypes` template parameter. If 100% of your existing instances are Spot Instances, switch some of them to On-Demand Instances by setting `OnDemandPercentage` parameter to a value above zero.
 
+## Instances are abruptly terminated
+
+This can happen when using Spot Instances. AWS EC2 sends a notification to a spot instance 2 minutes prior to termination. The agent intercepts that notification and attempts to gracefully shut down. If the instance does not shut down gracefully in that time, it is terminated.
```
Comment (jradtilbrook) [inline pages/agent/v3/elastic_ci_aws/troubleshooting.md:40]: @james2791 do you have an example of the log line we can document for people to use to know if their instance has been reclaimed?

Comment (jradtilbrook) [reply pages/agent/v3/elastic_ci_aws/troubleshooting.md:40]: Some ideas for improvements:
1. Document adding retries to the step so it automatically retries if reclaimed.
2. Document how to search the logs to know if an instance has been reclaimed.

Comment (jradtilbrook) [reply pages/agent/v3/elastic_ci_aws/troubleshooting.md:40]: Example: `| 2023-07-31 19:19:23.432 | level=info msg="Received termination notice" instanceId=i-abcd notice=spot | i-abcd | 444793955923:/buildkite/lifecycled |`

**Comment by jradtilbrook** [comment]
@mbelton-buildkite do you think we should merge this?

File: pages/agent/v3/elastic_ci_aws/troubleshooting.md [hunk 3]
```diff
@@ -35,6 +35,20 @@ Resource shortage can cause this issue. See the Auto Scaling group's Activity lo
...
+
+To identify if your agent instance was terminated, you can inspect the `/buildkite/lifecycled` CloudWatch log group for the instance. The example below shows the log line indicating that the instance was sent the spot termination notice.
+
+```
+| 2023-07-31 19:19:23.432 | level=info msg="Received termination notice" instanceId=i-abcd notice=spot | i-abcd | 444793955923:/buildkite/lifecycled |
+```
+
+If all your existing instances are Spot Instances, switch some of them to On-Demand Instances by setting the `OnDemandPercentage` parameter to a value above zero.
+
+For better resilience, you can utilise step retries to automatically retry a job that has failed due to spot instance reclamation. See [Automatic retry attributes](/docs/pipelines/command-step#retry-attributes-automatic-retry-attributes) for more information.
```
Comment (mbelton-buildkite) [inline pages/agent/v3/elastic_ci_aws/troubleshooting.md:50]: US spelling so `utilise` -> `utilize` but then both are just more complicated ways of saying `use`.

```suggestion
For better resilience, you can use step retries to automatically retry a job that has failed due to spot instance reclamation. See [Automatic retry attributes](/docs/pipelines/command-step#retry-attributes-automatic-retry-attributes) for more information.
```

**Review by mbelton-buildkite — APPROVED** [review]
Yip, let's get this live. I've left one last comment to address before you merge. :)

**Review by mbelton-buildkite — APPROVED** [review]

**Pull request merged**